apps/predictions/lib/repo.ex

```diff
@@ -188,20 +188,32 @@ defmodule Predictions.Repo do
 
     route = Routes.Repo.get(route_id)
 
-    [
-      %Predictions.Prediction{
-        id: id,
-        trip: trip,
-        stop: stop,
-        route: route,
-        direction_id: direction_id,
-        time: time,
-        stop_sequence: stop_sequence,
-        schedule_relationship: schedule_relationship,
-        track: track,
-        status: status,
-        departing?: departing?
-      }
-    ]
+    # For subway, drop predictions if the predicted time is earlier than the current time:
+    prediction_in_the_past =
+      if time == nil do
+        false
+      else
+        !Util.time_is_greater_or_equal(Util.to_local_time(time), Util.to_local_time(Util.now()))
+      end
+
+    if Route.subway?(route.type, route_id) && prediction_in_the_past do
```

Comment:
One red flag for me about doing this here is that this function is now doing two things. In addition to converting to a struct, it's also filtering (which isn't obvious where it is called). Would it be possible to add a new filter step below `filter_by_min_time`? At a glance I think your logic here would probably fit into the same structure as `filter_by_min_time`.